src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -397,14 +404,13 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
   //Steps: Delete every member from the underlying group and then add all of the new members. Do this in a *single*
   //transaction so if any bit fails, all of it fails and we don't end up in some incorrect intermediate state.
   private def overwritePolicyMembersInternal(id: FullyQualifiedPolicyId, memberList: Set[WorkbenchSubject])(implicit session: DBSession): Int = {
-    val p = PolicyTable.syntax("p")
     val gm = GroupMemberTable.syntax("gm")
 
-    val groupId = samsql"""delete from ${GroupMemberTable as gm}
-        using ${PolicyTable as p}
-        where ${gm.groupId} = ${p.groupId}
-        and ${p.name} = ${id.accessPolicyName}
-        and ${p.resourceId} = (${ResourceTable.loadResourcePK(id.resource)}) returning ${p.groupId}""".updateAndReturnGeneratedKey().apply()
+    val groupId = samsql"${workbenchGroupIdentityToGroupPK(id)}".map(rs => rs.long(1)).single().apply().getOrElse {
+      throw new WorkbenchException(s"Group for policy [$id] not found")
+    }
+
+    samsql"delete from ${GroupMemberTable as gm} where ${gm.groupId} = ${groupId}".update().apply()
```

Comment:
The old code was nice in that it deleted and returned the id in one round trip but it does not work when there are no members in the policy to start with. I reformulated select the id first, then delete. I thought this was more readable and reliable than handling an exception and doing a second query only on exception.